treematcher.py

```diff
@@ -307,116 +199,150 @@ def smart_lineage(self, constraint):
 
         return constraint
 
+class TreePattern(Tree):
+    def __str__(self):
+        return self.get_ascii(show_internal=True, attributes=["name"])
+
+
+    def __init__(self, newick=None, format=1, dist=None, support=None,
```

Comment:
I am defaulting to format 1 and quoted names, so instances do not require always the same args